lib/koans/19_streams.ex

```diff
@@ -0,0 +1,27 @@
+defmodule Streams do
+  use Koans
+
+  @intro "Streams"
+
+  koan "Streams are lazier than Enum, they only return a struct" do
+    assert Stream.map([1, 2, 3], &(&1 + 1)).__struct__ == ___
```

Comment:
As a learning, I wouldn't know what the form of the structure is here. It might also be nice to explain _why_ a structure is returned. Something like "Unlike Enum, a Stream does not execute immediately. It instead returns a struct the represents the enumeration to be performed later". I'm not a huge fan of that wording, but does is my point clear?

It might also be nice to initially have an example like "Streams iterate over collections and Enums" followed by something like "But they're lazy".